lib/rubocop/cop/betterment/non_standard_actions.rb

```diff
@@ -0,0 +1,72 @@
+module RuboCop
+  module Cop
+    module Betterment
+      class NonStandardActions < Cop
+        MSG_GENERAL = 'Use a new controller instead of custom actions.'.freeze
+        MSG_RESOURCE_ONLY = "Resource route refers to a non-standard action in it's 'only:' param. #{MSG_GENERAL}".freeze
+        MSG_ROUTE_TO = "Route goes to a non-standard controller action. #{MSG_GENERAL}".freeze
+
+        def_node_matcher :routes?, <<-PATTERN
+          (block (send (send (send (const nil? :Rails) :application) :routes) :draw) ...)
+        PATTERN
+
+        def_node_matcher :resource_with_only, <<-PATTERN
+          (send nil? {:resource :resources} _ (hash <(pair (sym :only) {(array (sym $_)*) (sym $_*)} ) ...> ))
+        PATTERN
+
+        def not_to_or_action?(sym)
+          !%i(to action).include?(sym)
+        end
+
+        def_node_matcher :route_to, <<~PATTERN
+          (send nil? {:match :get :post :put :patch :delete} ({str sym} $_) (hash {
+            <(pair (sym ${:to :action}) ({str sym} $_)) ...>
+            (pair (sym $#not_to_or_action?) $_)*
+          })?)
+        PATTERN
+
+        def on_block(node)
```

Comment:
what does on_block do versus something like on_send in the example on this [rubocop doc page](https://docs.rubocop.org/rubocop-ast/node_pattern.html)?